### ISSUE TITLE
Dockerfile.template uses latest version

### ISSUE THREAD

**Issue by sbiermann** [jpmeijers/ttn-resin-gateway-rpi#10]
Hi,
i think it would be better if in the Dockerfile.template is not the latest tag used. Because the idea of Docker is to have with the Dockerfile a reproduceable image which is every time you build it the same as before. If the latest tag is used, it will be build every time another image because of security updates of some packages. That makes it much easier to support.
If it is possible it would be nice when you can change it. 

I use your Dockerfile.template for creating my local docker image to avoid the installation nightmare with different raspbian versions and so on. It works great, good job!

Kind regards
Stefan

**Comment by jpmeijers** [comment]
Yes I see what you mean. The latest PR changed it to `latest`, and I don't know the reasoning behind it. There was a comment that the changes to the Dockerfile.template made an improvement to the build time.

Perhaps @kersing knows the reasoning behind using the `latest` tag.

**Comment by kersing** [comment]
Previously no tag was mentioned resulting in latest as well, this made it just more explicit. 

As the build script does not specify specific revisions of the code in the repositories builds will differ anyway. If you want a reproducible build you need to update both the template and dev/build.sh to pull specific versions of the code from git. (Moving the build process from dev/build.sh to the Dockerfile would be an improvement as well)

**Comment by kersing** [comment]
And if anyone wants to make a real contribution, switch to the alpine base image :smile: 
This won't be a walk in the park as Alpine Linux does not provide (minimal required versions of) some of the tools required to build the code.

**Comment by sbiermann** [comment]
I make some tests with alpine using as base image for the docker image. 

In the build process i have to add the new include to the loragw_spi.native.c because the _IOC_SIZEBITS was undefined. Which can be done by the build.sh script:
`sed -i '/ioctl.h>/#include <asm/ioctl.h> \' src/loragw_spi.native.c`

After that it compiles all stuff except the mp_pkt_forwarder from @kersing, which uses qsort_r  in the jitqueue.c and that is no part of alpine.
`obj/jitqueue.o: In function `jit_sort_queue':
jitqueue.c:(.text+0x154): undefined reference to `qsort_r'
collect2: error: ld returned 1 exit status
make: *** [Makefile:101: mp_pkt_fwd] Error 1
`
At that point it can not be fixed with a simple sed command as i have done it before. 
@kersing it is possible that you can add a simple switch for using qsort_s instead of qsort_r when using alpine linux?
